### PR TITLE
Title/web publica/ot105 48

### PR DESCRIPTION
**Pull request by ghost**
Summary
Added Title component and MUI library. Removed files related to CRA redux template.

User Story
https://alkemy-labs.atlassian.net/browse/OT105-48?atlOrigin=eyJpIjoiZjllYWRlNWQyMjI5NGUyYTliYjEwYjI4ZjQ4Njg0ZDUiLCJwIjoiaiJ9

Changes
- Removed CRA redux template files (features folder, logo.svg and some lines in store.js)
- Added emotion/react, emotion/styled and mui/material dependencies in package.json, required to run MUI library.
- Added 'Roboto' font in index.html, necessary for MUI to run properly.
- Added Title component, which receives 7 props. More details are commented in .js file.

Screenshot
image

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -24,6 +24,10 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
+    <link
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
+    />
```
Comment (saralavanchy) [inline public/index.html]: why this change is necessary?

File: package.json [hunk 2]
```diff
@@ -3,6 +3,9 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@emotion/react": "^11.6.0",
+    "@emotion/styled": "^11.6.0",
+    "@mui/material": "^5.2.1",
```
Comment (saralavanchy) [inline package.json]: this libraries where approved by your teammates?

File: src/App.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
-import logo from './logo.svg';
-import { Counter } from './features/counter/Counter';
+
 import './App.css';
+
```
Comment (saralavanchy) [inline src/App.js]: this empty space is not necessary

File: src/App.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
-import logo from './logo.svg';
-import { Counter } from './features/counter/Counter';
+
```
Comment (saralavanchy) [inline src/App.js]: this empty space is not necessary

File: src/App.js [hunk 5]
```diff
@@ -14,6 +14,7 @@ import ToysCampaign from './Campaigns/Toys/ToysCampaign';
 import MembersForm from './Components/Members/MembersForm';
 import ProjectsForm from './Components/Projects/ProjectsForm';
 
+
```
Comment (saralavanchy) [inline src/App.js]: this empty space is not necessary

File: src/App.js [hunk 6]
```diff
@@ -33,51 +34,7 @@ function App() {
...
-            className="App-link"
-            href="https://react-redux.js.org/"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            React Redux
-          </a>
-        </span>
-      </header>
+      Hi!
```
Comment (saralavanchy) [inline src/App.js]: please delete this unnecessary hi

File: src/Components/Title/Title.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import { Typography } from '@mui/material'
+
+//Title component receives 7 props: text, color, fontSize, imageUrl, bckgColor, bckgOpacity and paddingBlock
+//text (required) : The text of the title
+//color (optional): Sets the color text of the title. It receives a MUI color. Default: 'black'
+//variant (optional): Sets the MUI variant of the text. https://mui.com/api/typography/ for more information. Default: 'h2'
+//imageUrl (optional): The URL of the background image.
+//bckgColor (optional): This will only work if no imageUrl is provided. Sets the background color of the image. Default: #FAFA88 
+//bckgOpacity (optional): Sets the opacity of the background image. Default: '0.7'
+//paddingBlock (optional): Sets the padding-block of the Title component. Default: '6rem'
```
Comment (saralavanchy) [inline src/Components/Title/Title.js]: please delete this comments. the code shoudl be enought documentation to itself. One of the rules of clean code. If is not clear what the parameter do the parameter name is incorrect

File: src/Components/Title/Title.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+
+export default function Title({
+  text,
```
Comment (saralavanchy) [inline src/Components/Title/Title.js]: text is a too generic name, maybe titleText or title is better  name

File: src/Components/Title/Title.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+//variant (optional): Sets the MUI variant of the text. https://mui.com/api/typography/ for more information. Default: 'h2'
+//imageUrl (optional): The URL of the background image.
+//bckgColor (optional): This will only work if no imageUrl is provided. Sets the background color of the image. Default: #FAFA88 
+//bckgOpacity (optional): Sets the opacity of the background image. Default: '0.7'
+//paddingBlock (optional): Sets the padding-block of the Title component. Default: '6rem'
+
+
+export default function Title({
+  text, 
+  color = 'black',
```
Comment (saralavanchy) [inline src/Components/Title/Title.js]: color is too generic name, maybe titleTextColor or titleFontColor or titleColor its a better name

File: src/Components/Title/Title.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  variant = 'h2',
```
Comment (saralavanchy) [inline src/Components/Title/Title.js]: variant is too generic name, maybe titleMuiVariant is a better name

File: src/Components/Title/Title.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+//paddingBlock (optional): Sets the padding-block of the Title component. Default: '6rem'
+
+
+export default function Title({
+  text, 
+  color = 'black',
+  variant = 'h2', 
+  imageUrl, 
+  bckgOpacity = '0.7',
+  paddingBlock = '6rem'
```
Comment (saralavanchy) [inline src/Components/Title/Title.js]: maybe titlePadding

File: src/Components/Title/Title.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+}) {
+
+  const containerStyles = {
+    paddingBlock: paddingBlock, 
+    background: !imageUrl && '#FAFA88', 
+  }
+
+
```
Comment (saralavanchy) [inline src/Components/Title/Title.js:21]: this extra space is unnecessary

File: src/Components/Title/Title.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+export default function Title({
+  text, 
+  color = 'black',
+  variant = 'h2', 
+  imageUrl, 
+  bckgOpacity = '0.7',
+  paddingBlock = '6rem'
+}) {
+
```
Comment (saralavanchy) [inline src/Components/Title/Title.js]: this extra space is unnecessary

File: src/Components/Title/Title.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+//text (required) : The text of the title
+//color (optional): Sets the color text of the title. It receives a MUI color. Default: 'black'
+//variant (optional): Sets the MUI variant of the text. https://mui.com/api/typography/ for more information. Default: 'h2'
+//imageUrl (optional): The URL of the background image.
+//bckgColor (optional): This will only work if no imageUrl is provided. Sets the background color of the image. Default: #FAFA88 
+//bckgOpacity (optional): Sets the opacity of the background image. Default: '0.7'
+//paddingBlock (optional): Sets the padding-block of the Title component. Default: '6rem'
+
+
+export default function Title({
```
Comment (saralavanchy) [inline src/Components/Title/Title.js]: to follow the logic of the rest of the components please use const instead of function
```
const Title = ({....}) => {
...
}
export default Title;
```

File: src/Components/Title/Title.module.css [hunk 15]
```diff
@@ -0,0 +1,27 @@
+@font-face {
```
Comment (saralavanchy) [inline src/Components/Title/Title.module.css]: all the styles in a new Styles folder out of the Components routes: src/Styles/...

**Review by saralavanchy — CHANGES_REQUESTED** [review]

File: public/index.html (hunk 1 above)
Comment (ghost) [reply public/index.html]: Thought it was necessary in order for MUI to work

File: package.json (hunk 2 above)
Comment (ghost) [reply package.json]: We agreed on working with MUI. I don't know if @emotion/react and @emotion/styled are required to work with MUI. Just followed the instructions from the documentation.

File: src/App.js (hunk 3 above)
Comment (ghost) [reply src/App.js]: Will be removing empty spaces on next commit

**Review by saralavanchy — APPROVED** [review]

**Review by saralavanchy — APPROVED** [review]
